crates/db_views/src/community/community_follower_view.rs

```diff
@@ -78,18 +82,17 @@ impl CommunityFollowerView {
   pub async fn count_community_followers(
     pool: &mut DbPool<'_>,
     community_id: CommunityId,
-  ) -> Result<i64, Error> {
+  ) -> LemmyResult<i64> {
     let conn = &mut get_conn(pool).await?;
-    let res = Self::joins()
+    Self::joins()
       .filter(community_actions::community_id.eq(community_id))
       .select(count_star())
       .first::<i64>(conn)
-      .await?;
-
-    Ok(res)
+      .await
+      .with_lemmy_type(LemmyErrorType::NotFound)
```

Comment:
Why add explicit NotFound error in so many places? Diesel's NotFound error is automatically converted to LemmyError NotFound so this shouldnt be necessary at all.

Reply:
So that all of these can explicitly return lemmy errors instead of diesel ones.

Reply:
Part of this rework was to try to remove *all* diesel errors from schema and db_views, and return `LemmyResult` for all the functions. Seems like a bad idea to only use diesel results for only these ones; I'd rather have the return types uniformly be lemmy results.

Reply:
`?` does the conversion

Reply:
Unfortunately that'd require wrapping every result in `Ok(` .